lib/spack/spack/test/concretize.py

```diff
@@ -895,7 +913,9 @@ def test_conditional_variants_fail(self, bad_spec):
             ("py-extension3@1.0 ^python@3.5.1", ["patchelf@0.10"], []),
         ],
     )
-    def test_conditional_dependencies(self, spec_str, expected, unexpected):
+    def test_conditional_dependencies(self, spec_str, expected, unexpected, fuzz_dep_order):
```

Comment:
(nitpick) I would move this part of the docstring above:
```
The original concretizer can be sensitive to this, so we use this to ensure that it
is tested forwards and backwards.
```
here, to move the explanation closer to the only context using the fixture.